.github/workflows/c-cpp.yml

```
@@ -70,6 +74,7 @@ jobs:
           key: cache-${{ matrix.os }}-${{ matrix.config }}-${{ github.sha }}
           restore-keys: |
             cache-${{ matrix.os }}-${{ matrix.config }}-
+        if: matrix.os == 'ubuntu-latest'
```

```suggestion
        if: matrix.os != 'macos-latest'
```

The Windows CI also uses the cache action for the vcpkg build cache.

---

You modified the wrong line.